samples/drivers/video/capture/src/main.c

```diff
@@ -181,11 +183,15 @@ int main(void)
 
 	/* Get supported controls */
 	LOG_INF("- Supported controls:");
+	const char *dev_name = NULL;
+	struct video_ctrl_query cq = {.dev = video_dev, .id = VIDEO_CTRL_FLAG_NEXT_CTRL};
 
-	struct video_ctrl_query cq = {.id = VIDEO_CTRL_FLAG_NEXT_CTRL};
-
-	while (!video_query_ctrl(video_dev, &cq)) {
-		video_print_ctrl(video_dev, &cq);
+	while (!video_query_ctrl(&cq)) {
+		if (strcmp(dev_name, cq.dev->name) != 0) {
+			dev_name = cq.dev->name;
```

Comment:
[blocking :x:]
Can `NULL` be passed to `strcmp()`? I think `dev_name = NULL` on first iteration.

Would it work to compare the pointers?
```suggestion
		if (dev_curr != cq.dev) {
			dev_curr = cq.dev;
```

Maybe I misread the change though.

Reply:
Changed to compare the device rather than the device's string name which is much simpler.